internal/controller/promotion/argocd_test.go

```diff
@@ -175,6 +345,252 @@ func TestArgoCDPromote(t *testing.T) {
 	}
 }
 
+func TestArgoCDMustPerformUpdate(t *testing.T) {
+	testCases := []struct {
```

Comment:
@krancour I couldn't resist to show how I would write these tests compared to how they're typically done in the code base.